lib/routes/account.js

```diff
@@ -1212,13 +1224,30 @@ module.exports = function (
         }
       },
       handler: function (request, reply) {
-        var uid = request.payload.uid
+        var uidHex = request.payload.uid
+        var uid = Buffer(uidHex, 'hex')
         var code = Buffer(request.payload.code, 'hex')
         var service = request.payload.service || request.query.service
         var reminder = request.payload.reminder || request.query.reminder
 
+        // Because we have no session token on this endpoint, metrics context
+        // metadata was stashed against a synthesized token for the benefit of
+        // the activity events. This fake request object allows the correct
+        // metadata to be gathered when we emit the events.
+        var fakeRequestObject = {
+          auth: {
+            credentials: {
+              uid: uid,
+              id: request.payload.code
```

Comment:
Not in the scope of this PR, but seems like we could give [gather](https://github.com/mozilla/fxa-auth-server/blob/master/lib/metrics/context.js#L125) some smarts to pull out `code` if auth creds are not found. This could potentially help us not have to make fake request objects.